docs/datagrid/datagrid-v2-migration.md

```diff
@@ -0,0 +1,33 @@
+Upgrade DataGrid to v2
+===
+The core of DataGrid has been changed in the version 2.0. We increased the peformance, flexibility and added accessibility supoort. It means that API options have been changed and definition of styles or data source are different. We described all changes below. 
+
+### Data source
+In the newest DataGrid, the data source definition has been redesigned. It is a critical upgrade, so please double-check if everything is configured properly.
+
+DataGrid v1
+```js
+const grid = new DataGrid.DataGrid('container', {
+    dataTable: new DataGrid.DataTable({ columns })
+});
+```
+
+DataGrid v2
+```js
+const grid = DataGrid.dataGrid('container', {
+    table: new DataTable({
+        columns
+    })
+});
+```
+
+### Columns
+The `useHTML` parameter has been moved to the `columns` object.
+
+You can find more information in our [API Reference](https://api.highcharts.com/dashboards/typedoc/interfaces/DataGrid_DataGridOptions.IndividualColumnOptions.html#useHTML).
+
+### Styling
+The name of CSS classes have been also changed. We recommend to read our [article about styling](https://www.highcharts.com/docs/datagrid/style-by-css) the DataGrid 2.0.
```

Comment:
I would mention about moving all the important things to the CSS and they don't have any equivalent in the API. (e.g. row heights)